activesupport/lib/active_support/core_ext/module/attribute_accessors.rb

```diff
@@ -203,4 +203,57 @@ def mattr_accessor(*syms, instance_reader: true, instance_writer: true, instance
     mattr_writer(*syms, instance_writer: instance_writer, instance_accessor: instance_accessor, default: default, location: location)
   end
   alias :cattr_accessor :mattr_accessor
+
+  def singleton_attr_reader(*syms, instance_reader: true, instance_accessor: true, default: nil, location: nil)
+    raise TypeError, "singleton attributes should be defined directly on class, not singleton" if singleton_class?
+    location ||= caller_locations(1, 1).first
+
+    syms.each do |sym|
+      singleton_class.attr_reader(sym)
+
+      if instance_reader && instance_accessor
+        define_singleton_attr_instance_reader(sym)
+      end
+
+      sym_default_value = (block_given? && default.nil?) ? yield : default
+      instance_variable_set("@#{sym}", sym_default_value) unless sym_default_value.nil? && class_variable_defined?("@#{sym}")
```

Comment:
Should `class_variable_defined?` be `instance_variable_defined?`?

Reply:
Indeed, fixed.